### DISCUSSION TITLE
subcommands and --version command

### DISCUSSION THREAD

**Discussion by Tobi-De** [DanCardin/cappa#189] [Q&A] [answered]
Hi, how are you ? hope you doing great :)

Is this possible ? It seems the answer currently is no

```python
@cappa.command(
    help="Enhance your Django developer experience: CLI and Guides for the Modern Django Developer.",
)
class Falco:
    subcommand: cappa.Subcommands[StartProject | UpdateProject | Fmt | SyncDotenv]
    version: Annotated[
        bool,
        cappa.Arg(
            short="-v",
            long="--version",
            help="print cli version",
        ),
    ] = False

```

I feel like the only option is to implement version as a subcommand

**Comment by DanCardin** [comment, marked as answer]
you can do `cappa.parse/invoke(Falco, version='1.2.3')` (or `version=cappa.Arg('1.2.3', short="-v", long="--version", help="print cli version")` if the default arg definition isn't what you want).

I believe you should also be able to define it on the class itself if you prefer, by setting `action=ArgAction.version`. Although i personally wouldn't, because it doesn't really make sense as a normal class field, given that it will always just be the default value if not used.
